src/server/cookies.ts

```diff
@@ -299,12 +316,15 @@ export function getChunkedCookie(
 export function deleteChunkedCookie(
   name: string,
   reqCookies: RequestCookies,
-  resCookies: ResponseCookies
+  resCookies: ResponseCookies,
+  isLegacyCookie?: boolean
 ): void {
   // Delete main cookie
   resCookies.delete(name);
 
-  getAllChunkedCookies(reqCookies, name).forEach((cookie) => {
-    resCookies.delete(cookie.name); // Delete each filtered cookie
-  });
+  getAllChunkedCookies(reqCookies, name, isLegacyCookie ?? false).forEach(
```

Comment:
I believe we can avoid `?? false` here

```suggestion
  getAllChunkedCookies(reqCookies, name, isLegacyCookie).forEach(
```